threedi_api_qgis_client/utils.py

```diff
@@ -119,3 +119,19 @@ def check_download_checksum(download, filename):
         data = file_to_check.read()
         md5_returned = hashlib.md5(data).hexdigest()
         return download.etag == md5_returned
+
+
+def handle_api_exceptions(e):
+    """Extracting useful information from ApiException exceptions."""
+    error_body = e.body
+    try:
+        error_details = error_body["details"]
+    except TypeError:
+        error_body = json.loads(error_body)
+        errors = error_body["errors"]
+        error_parts = [f"{err['reason']} ({err['instance']['related_object']})" for err in errors]
+        error_details = "\n".join(error_parts)
+    except KeyError:
+        error_details = error_body
```

Comment:
I think a test would be real useful here. Obviously, you've fixed the code so that it gives a better error message in more situations.
Having a test for the previous (expected) kind of error and for the newly supported kind of error would be handy. Probably there will be other variants in the future: a test will make sure the ones you've fixed now *stay* fixed.